src/DSE.Open.Numerics/Sequence.Mean.cs

```diff
@@ -1,8 +1,26 @@
 // Copyright (c) Down Syndrome Education International and Contributors. All Rights Reserved.
 // Down Syndrome Education International and Contributors licence this file to you under the MIT license.
 
+using System.Numerics;
+
 namespace DSE.Open.Numerics;
 
 public static partial class Sequence
 {
+    public static TAcc Mean<T, TAcc>(ReadOnlySpan<T> sequence)
+        where T : struct, INumberBase<T>
+        where TAcc : struct, INumberBase<TAcc>
+    {
+        var sum = Sum<T, TAcc>(sequence);
+        return sum / TAcc.CreateChecked(sequence.Length);
+    }
+
+    public static TAcc Mean<T, TAcc>(IEnumerable<T> sequence)
```

Comment:
It's not hard to imagine giving a large sequence of `short`, wanting them accumulated into a `long` and wanting an `int` result:

```cs
public static TAccumulator Mean<TSource, TAccumulator, TResult>(IEnumerable<TSource> sequence)
    where TSource : struct, INumberBase<TSource>
    where TAccumulator : struct, INumberBase<TAccumulator>,
    where TResult : struct, INumberBase<TResult>
{
    TAccumulator sum = Sum<TSource, TAccumulator>(sequence, out var size);

    if (size == 0)
    {
        EmptySequenceException.Throw();
    }

    return TResult.CreateChecked(sum) / TResult.CreateChecked(size);
}
```

Reply:
Yup, though that is a simple cast or rounding function for the caller. Would we want the complexity of including those options here?

Reply:
Probably not public for now, but perhaps for the private implementation they're calling into.